### PR TITLE
Suggestion: Add a way to set slide cover images

### PR DESCRIPTION
**Pull request by mplewis**
Hey @jdan, thanks for writing Cleaver!

This PR is a suggestion more than a request to merge code. I wanted to add background cover images to some of my slides, so I did.

To add a cover image to a slide, just add `<!-- background: http://example.com/batmobile.jpg -->` to the body of any slide. It will generate the background CSS to set the background image for each individual slide. It will set the CSS background size for all slides to `cover`.

Let me know what you think. I'd be happy to re-implement this in a different way if you like the idea but not the code.

### CODE REVIEW

File: templates/bkgImages.mustache [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{{#bkgImages}}
+  #{{slideId}} {
+    background: url({{{bkgImage}}});
+  }
+{{/bkgImages}}
+
+{{#bkgImages}}
```
Comment (mplewis) [inline templates/bkgImages.mustache]: This is probably unnecessary. You could set it to apply to all slide-class divs instead of this.

**Comment by jdan** [comment]
Hey @mplewis, thanks for the PR.

I'm not quite sure I understand the need for this - since one can pretty easily add a background image by bringing in some CSS.

I'm hesitant to start generating CSS from the original markdown, because that may set an unfortunate precedent where folks also want things like `<!-- font-size: -->` or `<!-- color -->`. What do you think?

File: package.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
     "debug": "^2.1.1",
     "highlight.js": "~7.3.0",
     "js-yaml": "2.1.0",
+    "lodash": "^3.9.3",
```
Comment (mplewis) [inline package.json]: Being tired makes me make bad decisions

**Comment by mplewis** [comment]
When I built my last presentation with Cleaver, I used lots of cover images. I could have put the URLs in CSS and written

```css
#slide-1: {
  background-image: ...;
}
```

I felt this decoupled my cover image data from my slides. And if I changed the order or number of slides, then my cover image locations changed to different slides too.

I agree though: using HTML comments is messy. I like the idea of adding slide metadata, but I don't think that's the solution. YAML front matter seems like a neat way to go, but I don't know what the syntax for that would look like on a slide by slide basis.

**Comment by mplewis** [comment]
btw: This was inspired by [Shower](https://github.com/shower/shower/blob/master/index.html#L15-18), an HTML5 presentation engine which supports cover images.

**Comment by jdan** [comment]
You can also annotate slides like so :)

```
-- class1 class2

### Some slide 

...
```

And then in your css:

```
.class1 {
   /* ... */
}
```

**Comment by mplewis** [comment]
That's a good idea, thanks! I'll try that next time.

**Comment by jdan** [comment]
Let me know if it's super confusing, perhaps we could improve it.

**Comment by mplewis** [comment]
I think the only thing I would suggest is adding an example to the readme showing how you can add a cover image. Maybe add a presentation that uses a cover image too, to show off how that might work.

**Comment by jdan** [comment]
Mhm it's in the [readme](https://github.com/jdan/cleaver#markup) :)

> **#slide-N** (for example, *#slide-3*) allows you to identify a particular full-bleed slide by its position in the slideshow. It extends to the bounds of the page.

http://jdan.github.io/cleaver/#4 shows this off.

**Comment by mplewis** [comment]
Gaaaah I feel really bad now for totally missing that! Thanks again.

**Pull request closed without merging**